rflx/identifier.py

```diff
@@ -18,9 +19,13 @@ def __init__(
             self._parts = list(identifier.parts)
             self.location = location or identifier.location
         else:
-            assert False, f'unexpected identifier type "{type(identifier).__name__}"'
+            error.append(
+                f'unexpected identifier type "{type(identifier).__name__}"',
+                Subsystem.ID,
+                Severity.ERROR,
+            )
+            error.propagate()
```

Comment:
This should still be an assertion and not an exception. This error cannot happen if type checking is used, which is a general assumption for all of our code.

Reply:
Removed the effected commit and squashed the fixes to the previous change requests.